### PR TITLE
[Bug 1128948] Show "translate article" while requesting with English

### PR DESCRIPTION
**Pull request by safwanrahman**
I think this will fix the issue!
Contributor can localize a article by following pattern
{{ host }}/{{ locale }}/kb/{{ english_article_slug }}/edit

I believe this should fix the issue! @mythmon r?

### CODE REVIEW

**Comment by mythmon** [comment]
I think this change should work, but you need to add a test for it.

**Comment by safwanrahman** [comment]
@mythmon I cant understand why this Test is failing. not related my code. maybe a travis problem. Can you please check?

**Comment by mythmon** [comment]
That test is a timeout when trying to download the NPM dependencies. That's generally out of your control. I'll restart the build.

File: kitsune/wiki/tests/test_views.py [hunk 1]
```diff
@@ -369,6 +369,16 @@ def test_new_document_slugs(self, get_current):
                                                args=[data['slug']],
                                                locale='en-US'))
 
+    def test_translate_with_parent_locale_slug(self):
+        doc = document(locale=settings.WIKI_DEFAULT_LANGUAGE, save=True)
```
Comment (mythmon) [inline kitsune/wiki/tests/test_views.py]: New tests should prefer to use `DocumentFactory` if possible.

File: kitsune/wiki/tests/test_views.py [hunk 2]
```diff
@@ -369,6 +369,16 @@ def test_new_document_slugs(self, get_current):
...
+        trans_doc = document(parent=doc, locale='bn-BD', slug='bn-BD_slug', save=True)
+        url = reverse('wiki.edit_document', args=[doc.slug], locale=trans_doc.locale)
+        response = self.client.get(url)
+        assert response.status_code == 200
```
Comment (mythmon) [inline kitsune/wiki/tests/test_views.py]: Don't use bare assert statements, use the `eq_` helpfer from nose. It produces easier to debug tests.

**Comment by safwanrahman** [comment]
@mythmon updated the tests! 
Have a look! :)

**Comment by safwanrahman** [comment]
tests pass! :+1: :fireworks:

**Comment by mythmon** [comment]
Can you add a second negative test that tests the case where there is no matching document, both in the case that there is no English document, and in the case that there is no translated document under another slug?

**Comment by safwanrahman** [comment]
@mythmon Added another 2 tests. Thanks for this
Found a bug in my code while writing tests. So changed the code also. Now its totally perfect I believe
r?

**Comment by safwanrahman** [comment]
@mythmon r?

File: kitsune/wiki/views.py [hunk 3]
```diff
@@ -385,8 +385,25 @@ def steal_lock(request, document_slug, revision_id=None):
...
+        # Check if the document slug is available in default language.
+        parent_doc = get_object_or_404(Document, locale=settings.WIKI_DEFAULT_LANGUAGE,
+                                       slug=document_slug)
+        # If the document is available in default language, show the user the translation page
+        # of the requested locale
+        translation = parent_doc.translated_to(request.LANGUAGE_CODE)
+        # If the document is translated into the requested locale, show them the edit article
+        # page of that translated document
+        if translation:
+            doc = Document.objects.get(locale=request.LANGUAGE_CODE, slug=translation.slug)
```
Comment (mythmon) [inline kitsune/wiki/views.py]: This seems like it is just re-fetching the same document from DB again. How is `Document.objects.get(locale=request.LANGUAGE_CODE, slug=translation.slug)` different from `translation`?

Comment (safwanrahman) [inline kitsune/wiki/views.py]: Oh! I did not notice it! I just gone by as usual I do to fetch a document!
Thanks :+1: 
I am fixing it!

**Comment by safwanrahman** [comment]
@mythmon Fixed the issue and squashed it into one!
This is ready to merge!

**Comment by mythmon** [comment]
Looks good. Once Travis passes, I'll merge this.

**Pull request merged**